src/generate_rpkg_from.cpp

```diff
@@ -297,7 +297,13 @@ void rpkg_function::generate_rpkg_from(std::string& input_path, std::string& out
 
             uint64_t compressed_size_final;
 
-            compressed_size_final = LZ4_compress_HC(input_file_data.data(), output_file_data.data(), static_cast<int>(input_file_size), compressed_size, LZ4HC_CLEVEL_MAX);
+            if (compress)
```

Comment:
Surely this shouldn't compress at all if it is false?

And if that's not what the variable is supposed to name, rename it, it's very unclear.